newdle/client/src/components/answer/AnswerPage.js

```diff
@@ -10,13 +10,59 @@ import {
   getNewdle,
   getNumberOfAvailableAnswers,
   getNumberOfTimeslots,
+  getParticipant,
   isAllAvailableSelected,
   isAllAvailableSelectedImplicitly,
 } from '../../answerSelectors';
-import {chooseAllAvailable} from '../../actions';
+import {abortAnswering, chooseAllAvailable, fetchParticipant} from '../../actions';
 import styles from './answer.module.scss';
 import client from '../../client';
 
+function ParticipantName({anonymous, setName, onSubmit, disabled}) {
+  const dispatch = useDispatch();
+  const {partcode: participantCode, code: newdleCode} = useParams();
+  const participant = useSelector(getParticipant);
+
+  useEffect(() => {
+    if (!anonymous) {
+      dispatch(fetchParticipant(newdleCode, participantCode));
+      return () => {
+        dispatch(abortAnswering());
```

Comment:
I don't think we need this here again; we already do it in `AnswerHeader`.

However, I think fetching from the header isn't super clear (I just did it there since we do the same on the summary page).
What about having some kind of `AnswerController` component that doesn't render anything but takes care of fetching/aborting? This component would then be mounted through yet another `<Route>` matching th answer URL.

Reply:
Agreed. Not something I'd do in this PR, though.